### PR TITLE
Lobby xeno late join queue information

### PR DESCRIPTION
**Pull request by Drulikar**
# About the pull request

This PR adds some xeno queue information when late joining as a xeno from the lobby. It also streamlines the late join xeno from lobby process somewhat (automatically offers observation if there are no afk xenos, and automatically turns on xeno candidacy preference if done so). It also adds the same position estimate to your first cached message if you were to press the join as xeno button when you are presently ineligible.

There is also misc refactoring and additional autodoc comments for related variables.

# Explain why it's good for the game

Better new player experience and a bit more information for a player on the lobby to determine whether they want to wait in the xeno queue or late join as a marine.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

original implementation:
example

Addition to ineligible blub:
image

tgui_alert change:
example2

</details>

# Changelog
:cl: Drathek
add: Added xeno queue information when attempting a xeno late join while inelgible (from the lobby and first time after death)
qol: Streamlined the xeno late join process from lobby (offers observation and sets xeno preference)
ui: Changed some alerts on main menu to tgui_alert
/:cl:

### CODE REVIEW

**Comment by ihatethisengine** [comment]
now let me see the same number as a xeno who just died

**Comment by Drulikar** [comment]
> now let me see the same number as a xeno who just died

Ok. 
image

File: code/game/gamemodes/cm_initialize.dm [hunk 1]
```diff
@@ -389,15 +389,39 @@ Additional game mode variables.
...
+				// No cached/current lobby message, determine the position
+				var/list/valid_candidates = get_alien_candidates()
+				var/candidate_time = candidate_new_player.client.player_details.larva_queue_time
+				var/position = 1
+				for(var/mob/dead/observer/current in valid_candidates)
+					if(current.client.player_details.larva_queue_time >= candidate_time)
+						break
+					position++
+				candidate_new_player.larva_queue_message_stale_time = world.time + 3 MINUTES // spam prevention
+				candidate_new_player.larva_queue_cached_message = "Your position would be [position]\th in the larva queue if you observed and were elgible to be a xeno. \
```
Comment (harryob) [inline code/game/gamemodes/cm_initialize.dm:413]:
```suggestion
				candidate_new_player.larva_queue_cached_message = "Your position would be [position]\th in the larva queue if you observed and were eligible to be a xeno. \
```

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -232,6 +211,48 @@
...
+	observer.set_lighting_alpha_from_pref(client)
+	spawning = TRUE
+	observer.started_as_observer = TRUE
+
+	close_spawn_windows()
+
+	var/obj/effect/landmark/observer_start/O = SAFEPICK(GLOB.observer_starts)
+	if(istype(O))
+		to_chat(src, SPAN_NOTICE("Now teleporting."))
+		observer.forceMove(O.loc)
```
Comment (harryob) [inline code/modules/mob/new_player/new_player.dm:235]:
```suggestion
	var/obj/effect/landmark/observer_start/spawn_point = SAFEPICK(GLOB.observer_starts)
	if(istype(spawn_point))
		to_chat(src, SPAN_NOTICE("Now teleporting."))
		observer.forceMove(spawn_point.loc)
```

File: code/modules/mob/new_player/new_player.dm [hunk 3]
```diff
@@ -232,6 +211,48 @@
...
+		return
+	if(!client)
+		return
+	if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
+		to_chat(src, SPAN_WARNING("The game is still setting up, please try again later."))
+		return
+
+	if(!client.prefs?.preview_dummy)
+		client.prefs.update_preview_icon()
+	var/mob/dead/observer/observer = new /mob/dead/observer(get_turf(pick(GLOB.latejoin)), client.prefs.preview_dummy)
```
Comment (harryob) [inline code/modules/mob/new_player/new_player.dm:225]:
```suggestion
	var/mob/dead/observer/observer = new(get_turf(pick(GLOB.latejoin)), client.prefs.preview_dummy)
```

**Review by harryob — CHANGES_REQUESTED** [review]
nits

**Review by harryob — APPROVED** [review]

**Pull request merged**